stellargraph/data/explorer.py

```diff
@@ -796,3 +800,202 @@ def _check_neighbourhood_sizes(self, in_size, out_size):
+                edge_time_label: <string> Label of the edge weight property.
+           """
+
+        if type(bidirectional) != bool:
+            self._raise_error(
+                "Parameter bidirectional has to be either False (temporal random walks forward in times) or True (temporal random walks extending in both directions)."
+            )
+
+        if not isinstance(edge_time_label, str):
+            self._raise_error("The edge time property label has to be of type string")
+
+        "Check that all edges have the time label specified by 'edge_time_label' and all the time values positive real values."
+        for node in self.graph.nodes():
+
+            for neighbor in self.graph.neighbors(
+                node, include_edge_weight=True, edge_types=[edge_time_label]
+            ):
+                for n, t in neighbor:
+                    if not isinstance(t, (int, float)):
+                        self._raise_error(
+                            "Timestamp between nodes ({}) and ({}) is not numeric ({}).".format(
+                                node, neighbor, t
+                            )
+                        )
+                    elif t < 0:  # check if edge has a negative timestamp
+                        self._raise_error(
+                            "Timestamp between nodes ({}) and ({}) is negative ({}).".format(
+                                node, neighbor, t
+                            )
+                        )
+
+    def get_exp_distribution(self, times, current_time, orientation="forward"):
+        min_time = min(times)
+        time_dist = [
+            np.exp(-(np.abs(t - current_time) - np.abs(min_time - current_time)))
+            for t in times
+        ]  # relative gap in time w.r.t current time (higher values for shorter gaps)
+        sum_time_dist = sum(time_dist)
+
+        exp_dist = [t / sum_time_dist for t in time_dist]  # exponential distribution
+        return exp_dist
+
+    def _step(self, node, time, is_forward, edge_time_label, step_type, rs):
+        """Perform 1 temporal step from a node. Returns None if a dead-end is reached."""
+
+        def check_time(t):
+            return (is_forward and t > time) or (not is_forward and t < time)
+
+        edges = [
+            (neighbour, t)
+            for neighbour, t in self.graph.neighbors(
+                node, include_edge_weight=True, edge_types=[edge_time_label]
+            )
+            if check_time(t)
+        ]
+
+        if len(edges) > 0:
+            if step_type == "uniform_random":
+                next_edge = rs.sample(edges, 1)
+                return next_edge[0][0], next_edge[0][1]
+            else:
+                neighbors, times = zip(*edges)
+                dist = self.get_exp_distribution(
+                    times, time, "forward"
+                )  # exponential distribution
+                chosen_time = choice(times, 1, p=dist)[0]
+                next_edge = edges[times.index(chosen_time)]
```

Comment:
I think this is not very efficient `O(len(times))` and also problematic if the same time value occurs multiple times - will think of another way